### PR TITLE
Major release 9

### PR DESCRIPTION
**Pull request by hannahramadan**
Includes all changes for the 9.0 release

[All PRs included in this branch](https://github.com/newrelic/newrelic-ruby-agent/pulls?q=is%3Apr+is%3Amerged+base%3Amajor-release-9+)

### CODE REVIEW

**Comment by tannalynn** [comment]
[ci cron run](https://github.com/newrelic/newrelic-ruby-agent/actions/runs/4167144248)
[jruby run](https://github.com/newrelic/newrelic-ruby-agent/actions/runs/4167146260)

File: .github/workflows/ci_cron.yml [hunk 1]
```diff
@@ -310,40 +304,40 @@ jobs:
...
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: voxmedia/github-action-slack-notify-build@v1
-        if: ${{ env.WORKFLOW_CONCLUSION == 'success' && steps.last_status.outputs.last_status == 'failure' && github.event_name != 'workflow_dispatch' }}
-        env:
-          SLACK_BOT_TOKEN: ${{ secrets.RUBY_GITHUB_ACTIONS_BOT_WEBHOOK }}
-        with:
-          channel: ruby-agent-notifications
-          status: SUCCESS
-          color: good
+  # notify_slack_fail:
```
Comment (hannahramadan) [inline .github/workflows/ci_cron.yml:307]: I suppose we will want to reenable Slack notifications! I can make this PR if you want @tannalynn

**Review by tannalynn — APPROVED** [review]

File: lib/new_relic/agent/instrumentation/grape.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 DependencyDetection.defer do
   # Why not just :grape? newrelic-grape used that name already, and while we're
   # not shipping yet, overloading the name interferes with the plugin.
-  named :grape_instrumentation
+  @name = :grape_instrumentation
```
Comment (hannahramadan) [inline lib/new_relic/agent/instrumentation/grape.rb:12]: Why did we make this change to select instrumentation and not others?

Comment (tannalynn) [reply lib/new_relic/agent/instrumentation/grape.rb:12]: This was related to the [configuration options we removed](https://github.com/newrelic/newrelic-ruby-agent/pull/1782). Several of these are called `something_instrumentation`, but our configuration options are `instrumentation.something`. We have tests that expect configuration options called `instrumentation.something_instrumentation` when `named` is used, so we had to switch it to just set the name attribute directed.

File: test/environments/rails60/config/initializers/test.rb [hunk 3]
```diff
@@ -16,5 +16,6 @@ def sniff
 
   add_method_tracer :sniff
 end
-
+puts Rails.application.config.active_record.timestamped_migrations
```
Comment (hannahramadan) [inline test/environments/rails60/config/initializers/test.rb:19]: Leftover `puts` statements?

Comment (tannalynn) [reply test/environments/rails60/config/initializers/test.rb]: removed 4c9ccfe25d0d8180fd32d6bb510a8d696390687e

**Review by tannalynn — APPROVED** [review]

File: test/multiverse/suites/thread/Envfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+instrumentation_methods :chain, :prepend
+
+gemfile nil
```
Comment (hannahramadan) [inline test/multiverse/suites/thread/Envfile:7]: What is this envfile doing?

Comment (tannalynn) [reply test/multiverse/suites/thread/Envfile:7]: Not much lol. So we need to test the chain an prepend instrumentation, but since this is for the thread and fiber classes, there are no gems we need to install to test this, so we pass nil so it only creates a gemfile with the base multiverse gems we need.

Comment (hannahramadan) [reply test/multiverse/suites/thread/Envfile:7]: Cool, that makes sense :)

Comment (fallwith) [reply test/multiverse/suites/thread/Envfile:7]: This code (equivalent to `gemfile(nil)`) is invoking the `gemfile` method defined in `multiverse/lib/multiverse/envfile.rb` with `nil` as the argument.

It looks like doing so would accomplish the same as removing this `gemfile nil` line entirely. Perhaps it should be removed.

Comment (fallwith) [reply test/multiverse/suites/thread/Envfile:7]: Previously I was operating under the assumption that a `gemfile` call was needed to get a `Gemfile.1` file created. But given this line in the `gemfile` method:

```
@gemfiles.push(content) unless content.nil? || content.empty?
```

that doesn't appear to be the case.

File: test/new_relic/agent/tracer_test.rb [hunk 5]
```diff
@@ -359,10 +359,34 @@ def test_current_segment_in_nested_threads_auto
...
+        with_config(:'instrumentation.thread.tracing' => false) do
+          assert_nil Tracer.current_segment
+
+          txn = Tracer.start_transaction(name: "Controller/blogs/index", category: :controller)
+
+          assert_equal txn.initial_segment, Tracer.current_segment
+          threads = []
+
+          threads << ::Thread.new do
+            # nothong
```
Comment (hannahramadan) [inline test/new_relic/agent/tracer_test.rb:382]: lol

**Review by hannahramadan — COMMENTED** [review]
Looks good!
